### PR TITLE
 webui: port deprecated Wizard component to new PF5 implementation 

### PR DESCRIPTION
**Pull request by adamkankovsky**
In the end, the changes were a bit more extensive than I expected, but I tried to keep as much clarity and as much functionality as possible.

### CODE REVIEW

File: ui/webui/src/components/AnacondaWizard.jsx [hunk 1]
```diff
@@ -166,93 +157,112 @@ export const AnacondaWizard = ({ dispatch, isBootIso, osRelease, storageData, lo
...
+        if (prevStep.prevId !== currentStep.id) {
             // first reset validation state to default
             setIsFormValid(false);
         }
 
         // Reset the applied partitioning when going back from review page
-        if (prevStep.prevId === "installation-review" && newStep.id !== "installation-progress") {
+        if (prevStep.prevId === "installation-review" && currentStep.id !== "installation-progress") {
             setIsInProgress(true);
             resetPartitioning()
```
Comment (rvykydal) [inline ui/webui/src/components/AnacondaWizard.jsx:222]: Seems that resetPartitioning is not called when moving from installation-review back. Which corresponds with failing TestBasic.testSidebarNavigation.

Comment (rvykydal) [reply ui/webui/src/components/AnacondaWizard.jsx:222]: Ah, you need to replace `prevId` with `id`

Comment (rvykydal) [reply ui/webui/src/components/AnacondaWizard.jsx:222]: Ah one more thing probably related also to one of the failing pixel tests: TestStorageMountPoints-testBasic-mount-point-mapping-table. When you go back (either by Back or in sidebar steps, for example to Installation Method) the next steps (for example "Review and Install") are no more disabled (which they used to be).

File: ui/webui/src/components/AnacondaWizard.jsx [hunk 2]
```diff
@@ -166,93 +157,112 @@ export const AnacondaWizard = ({ dispatch, isBootIso, osRelease, storageData, lo
...
-            let step = ({
-                id: s.id,
+        return stepsOrder.filter(s => !s.isHidden).map(s => {
+            let stepProps = {
+                ariaLabel: `${title} content`,
                 name: s.label,
+                id: s.id,
                 stepNavItemProps: { id: s.id },
+                ...(s.isExpandable && { isExpandable: true }),
                 canJumpTo: canJumpToStep(s.id, currentStepId),
```
Comment (rvykydal) [inline ui/webui/src/components/AnacondaWizard.jsx:185]: canJumpTo seems to be removed in PF5, I guess we'll need to use isDisabled instead and the logic used in `canJumpToStep` will need to be modified a bit.

File: ui/webui/src/components/AnacondaWizard.jsx [hunk 3]
```diff
@@ -57,6 +55,7 @@ export const AnacondaWizard = ({ dispatch, isBootIso, osRelease, storageData, lo
     const [isFormValid, setIsFormValid] = useState(false);
     const [stepNotification, setStepNotification] = useState();
     const [isInProgress, setIsInProgress] = useState(false);
+    const [isInstallationProgress, setInstallationProgress] = useState(false);
```
Comment (KKoukiou) [inline ui/webui/src/components/AnacondaWizard.jsx:58]: There is no need (it's weird) for this to be state variable. It should just be constant.

const inInstallationProgressStep = path[0] === "installation-progress";

File: ui/webui/src/components/AnacondaWizard.jsx [hunk 4]
```diff
@@ -130,160 +130,143 @@ export const AnacondaWizard = ({ dispatch, isBootIso, osRelease, storageData, lo
             data: {
                 deviceData: storageData.devices,
                 diskSelection: storageData.diskSelection,
-                requests: storageData.partitioning ? storageData.partitioning.requests : null,
```
Comment (KKoukiou) [inline ui/webui/src/components/AnacondaWizard.jsx:133]: Thanks for shorting alphabetically the properties here and in the imports, but please do that in seperate commit if possible for code readability purposes.

File: ui/webui/src/components/AnacondaWizard.jsx [hunk 5]
```diff
@@ -130,160 +130,143 @@ export const AnacondaWizard = ({ dispatch, isBootIso, osRelease, storageData, lo
...
                 language,
-                osRelease
```
Comment (KKoukiou) [inline ui/webui/src/components/AnacondaWizard.jsx:135]: osRelease here was removed? Is that on the purpose?

File: ui/webui/src/components/AnacondaWizard.jsx [hunk 6]
```diff
@@ -130,160 +130,143 @@ export const AnacondaWizard = ({ dispatch, isBootIso, osRelease, storageData, lo
...
+            return (
+                <WizardStep key={s.id} {...stepProps} />
+            );
         });
-        return steps;
     };
-    const steps = createSteps(stepsOrder);
+    const steps = createSteps(stepsOrder, componentProps);
 
-    const goToStep = (newStep, prevStep) => {
```
Comment (KKoukiou) [inline ui/webui/src/components/AnacondaWizard.jsx:222]: Do you mind keeping the variable naming how is was before, just to cause less diff in the commit? Let's limit the commit to the required changes. Any improvements can come if you like in follow up, and be discussed separately.

File: ui/webui/src/components/review/ReviewConfiguration.jsx [hunk 7]
```diff
@@ -200,7 +200,8 @@ export const ReviewConfigurationConfirmModal = ({ idPrefix, onNext, setNextWaits
                 key="confirm"
                 onClick={() => {
                     setNextWaitsConfirmation(false);
-                    onNext();
+                    setInstallationProgress(true);
+                    window.location.hash = "#/installation-progress";
```
Comment (KKoukiou) [inline ui/webui/src/components/review/ReviewConfiguration.jsx:204]: Please use cockpit.location.go instead of writing to the window.location.hash object.

**Comment by garrett** [comment]
Being able to figure out where you are in a process and estimate how much longer you have left is one of core functions in a Wizard.

According to Nielsen Norman Group (well respected team who have written many of the fundamental books in the realm of software design): https://www.nngroup.com/articles/wizards/ — section "_Recommendations for Designing Usable Wizards_", subsection 2:

"**Communicate a clear mental model of the process** by displaying a list or a diagram of the steps involved and highlighting the current step.

Because the process is presented one step at a time, there is danger that users will miss the context, get confused, or not realize how long the progression will be. It best to set the right expectations and explain to the use what the process looks like and how many steps are involved.

Especially with complicated processes, a clear labeling of each of the steps as well as a clear indicator of the current step can help keep users oriented."

**Pull request closed without merging**